src/python_testing/TC_MCORE_FS_1_4.py

```diff
@@ -0,0 +1,482 @@
+
+        self.wait_for_text_event = threading.Event()
+        self.wait_for_text_text = None
+
+        args = [self.APP_PATH]
+        args.append(f"--app-admin={self.FABRIC_ADMIN_PATH}")
+        args.append(f"--app-bridge={self.FABRIC_BRIDGE_PATH}")
+        # Override default ports, so it will be possible to run
+        # our TH_FSA alongside the DUT_FSA during CI testing.
+        args.append("--app-admin-rpc-port=44000")
+        args.append("--app-bridge-rpc-port=44001")
+        # Keep the storage directory in a temporary location.
+        args.append(f"--storage-dir={storageDir}")
+        # FIXME: Passing custom PAA store breaks something
+        # if paaTrustStorePath is not None:
+        #     args.append(f"--paa-trust-store-path={paaTrustStorePath}")
+        if fabricName is not None:
+            args.append(f"--commissioner-name={fabricName}")
+        if nodeId is not None:
+            args.append(f"--commissioner-node-id={nodeId}")
+        args.append(f"--commissioner-vendor-id={vendorId}")
+        args.append(f"--secured-device-port={bridgePort}")
+        args.append(f"--discriminator={bridgeDiscriminator}")
+        args.append(f"--passcode={bridgePasscode}")
+
+        self.admin = Subprocess(args, stdout_cb=self._process_admin_output)
+        self.wait_for_text_text = "Connected to Fabric-Bridge"
+        self.admin.start()
+
+        # Wait for the bridge to connect to the admin.
+        self.wait_for_text()
+
+    def CommissionOnNetwork(self, nodeId, setupPinCode=None, filterType=None, filter=None):
+        self.wait_for_text_text = f"Commissioning complete for node ID 0x{nodeId:016x}: success"
+        # Send the commissioning command to the admin.
+        self.admin.p.stdin.write(f"pairing onnetwork {nodeId} {setupPinCode}\n")
+        self.admin.p.stdin.flush()
+        # Wait for success message.
+        self.wait_for_text()
+
+    def stop(self):
+        self.admin.stop()
+
+
+class AppServer:
+
+    def __init__(self, app, storageDir, port=None, discriminator=None, passcode=None):
+
+        args = [app]
+        args.extend(["--KVS", tempfile.mkstemp(dir=storageDir, prefix="kvs-app-")[1]])
+        args.extend(['--secured-device-port', str(port)])
+        args.extend(["--discriminator", str(discriminator)])
+        args.extend(["--passcode", str(passcode)])
+        self.app = Subprocess(args, tag="SERVER")
+        self.app.start()
+
+    def stop(self):
+        self.app.stop()
+
+
+class TC_MCORE_FS_1_4(MatterBaseTest):
+
+    @property
+    def default_timeout(self) -> int:
+        # This test has some manual steps, so we need a longer timeout.
+        return 200
+
+    def setup_class(self):
+        super().setup_class()
+
+        # Get the path to the TH_FSA (fabric-admin and fabric-bridge) app from
+        # the user params or use the default path.
+        FabricSyncApp.APP_PATH = self.user_params.get("th_fsa_app_path", FabricSyncApp.APP_PATH)
```

Comment:
I have always been under the impression that in python THING_IN_ALL_CAPS are supposed to indicate they constants. I would rather see this get assigned to lower case value.

I would rather see this passed into `FabricSyncApp` constructor, where the default if not provide is None we use the constant